### DISCUSSION TITLE
Virtual Keyboard/NumPad ?

### DISCUSSION THREAD

**Discussion by Den4ik1992** [Budibase/budibase#9342] [Help] [answered]
Hi guys, I'm brand new to buibase.
I would like to build a NumPad with several buttons to enter numbers in a "Text Field" via a touch monitor without a keyboard.
If I do it via OnClick - >DefineActions -> UpdateFieldValue the number in the text field is replaced but the number has to be appended on the value in the Text Field. 
A button to delete the last character would also be great ! 

is there a function to include your own html/css/javascript then i could just copy the project into mine for example.
https://codepen.io/coskuntekin/pen/jRQrpm

I would appreciate any help

**Comment by andz-bb** [comment, marked as answer]
hey @Den4ik1992 

this sounds like a great idea for a [custom component](https://docs.budibase.com/docs/custom-component) but if you want to do it with pure Budibase components, it should be possible.

It's a bit hacky but here's an example I made:

![number input demo](https://user-images.githubusercontent.com/110921612/213784515-daded4ee-99f7-4255-b10a-74dbe70e3e20.gif)

I put the numbers 0-9 into a table and then used a repeater to render each number button. The On click action for each of the buttons appends that button's number to the current field value. This is the JS function I use to calculate the new field value:

```
const existing = $("New Form.Fields.number") == null ? "" : $("New Form.Fields.number")
return existing + $("New Repeater.Numbers.number")
```

I've attached the example app I've made so you can have a closer look. There are definitely some limitations to this approach and with my demo specifically, but it shows what you can do just with the standard component functionality. Hope this helps!

[Numpad-export-1674243675107.tar.gz](https://github.com/Budibase/budibase/files/10469671/Numpad-export-1674243675107.tar.gz)

**Reply by Den4ik1992** [reply]
Thank you for your time and effort, without your help I would not have been able to do it.
